### PR TITLE
irssi: add HEAD

### PR DESCRIPTION
**Pull request by dunn**

### CODE REVIEW

File: Library/Formula/irssi.rb [hunk 1]
```diff
@@ -21,10 +27,19 @@ class Irssi < Formula
...
   depends_on "dante" => :optional
 
-  # Fix Perl build flags and paths in man page
+  # Make paths in man page Homebrew-specific
   patch :DATA
 
   def install
+    # Fix Perl build flags
+    conf = build.head? ? "configure.ac" : "configure"
+    inreplace conf do |s|
```
Comment (dunn) [inline Library/Formula/irssi.rb]: Was this a Mac- or Homebrew-specific patch? If it's not I can send it upstream.

Comment (DomT4) [inline Library/Formula/irssi.rb]: No idea. It was added in [2009](https://github.com/homebrew/homebrew/commit/8d6720204d28a49903058de2d289c6d6b9afedec). It looks like it was intended to be Homebrew specific.

Comment (mikemcquaid) [inline Library/Formula/irssi.rb]: I think it'd be good to try and either remove it or get it upstream.

Comment (dunn) [inline Library/Formula/irssi.rb]: I can build and run Irssi (with Perl stuff) without it, so I removed it.

Comment (DomT4) [inline Library/Formula/irssi.rb]: The difference in output between patch and no patch is:

```diff
- -arch x86_64 -arch i386 -g -pipe -fno-common -DPERL_DARWIN -fno-strict-aliasing -fstack-protector  -I/System/Library/Perl/5.18/darwin-thread-multi-2level/CORE %
+ -g -pipe -fno-common -DPERL_DARWIN -fno-strict-aliasing -fstack-protector  -I/System/Library/Perl/5.18/darwin-thread-multi-2level/CORE
```

And:

```diff
- -arch x86_64 -arch i386 -fstack-protector  -L/System/Library/Perl/5.18/darwin-thread-multi-2level/CORE -lperl
+ -fstack-protector  -L/System/Library/Perl/5.18/darwin-thread-multi-2level/CORE -lperl
```

Comment (mikemcquaid) [inline Library/Formula/irssi.rb]: So it's trying to build universally unconditionally? I think `superenv` will actually handle this for us now. Mind checking that out, @DomT4 (i.e. seeing if you remove the patch if it makes a universal binary)?

Comment (DomT4) [inline Library/Formula/irssi.rb]: No problem here:

```
checking build system type... x86_64-apple-darwin14.4.0
checking host system type... x86_64-apple-darwin14.4.0
```

```
~> file /usr/local/Cellar/irssi/0.8.17_2/bin/*
/usr/local/Cellar/irssi/0.8.17_2/bin/botti: Mach-O 64-bit executable x86_64
/usr/local/Cellar/irssi/0.8.17_2/bin/irssi: Mach-O 64-bit executable x86_64
```

File: Library/Formula/irssi.rb [hunk 2]
```diff
@@ -21,7 +27,7 @@ class Irssi < Formula
   depends_on "openssl" => :recommended
   depends_on "dante" => :optional
 
-  # Fix Perl build flags and paths in man page
+  # Make paths in man page Homebrew-specific
```
Comment (mikemcquaid) [inline Library/Formula/irssi.rb]: Can you switch this to using an `inreplace` instead (and ideally file an upstream issue to complain about `configure` not processing this file correctly)? Thanks!

**Comment by dunn** [comment]
https://github.com/irssi/irssi/issues/251

**Comment by DomT4** [comment]
This looks :100: now if Mike is happy too.

**Comment by dunn** [comment]
Should I remove the description?

**Comment by DomT4** [comment]
:+1:, Thanks!

**Comment by DomT4** [comment]
Merged in https://github.com/Homebrew/homebrew/commit/5ac6367565a116e1823e27cf1b6c7a7e314c5f39. Thanks @dunn! :rocket:

**Pull request closed without merging**